bode/bode/utils/graph.py

```diff
@@ -0,0 +1,53 @@
+from collections import defaultdict
+
+
+class FindUnion:
```

Comment:
What do you think about renaming this to `DisjointSets`? It should read a bit nicer when using it:
```python
disjoint_sets = DisjointSets(x)
disjoint_sets.find(x)
disjoint_sets.union(x, y)
```